NPE when running scalafix:scalafix

```
[ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin:0.1.1_0.9.5:scalafix (default-cli) on project mvn-scalafmt_2.12: Execution default-cli of goal io.github.evis:scalafix-maven-plugin:0.1.1_0.9.5:scalafix failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.github.evis:scalafix-maven-plugin:0.1.1_0.9.5:scalafix (default-cli) on project mvn-scalafmt_2.12: Execution default-cli of goal io.github.evis:scalafix-maven-plugin:0.1.1_0.9.5:scalafix failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.github.evis:scalafix-maven-plugin:0.1.1_0.9.5:scalafix failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at io.github.evis.scalafix.maven.plugin.params.PluginsParam$.compilerArgs (PluginsParam.scala:32)
    at io.github.evis.scalafix.maven.plugin.params.PluginsParam$.$anonfun$apply$3 (PluginsParam.scala:13)
    at scala.collection.immutable.List.flatMap (List.scala:338)
    at io.github.evis.scalafix.maven.plugin.params.PluginsParam$.$anonfun$apply$1 (PluginsParam.scala:13)
    at io.github.evis.scalafix.maven.plugin.ScalafixArgumentsBuilder.patch (ScalafixArgumentsBuilder.scala:77)
    at io.github.evis.scalafix.maven.plugin.phases.BuildScalafixArguments.$anonfun$buildScalafixArguments$1 (BuildScalafixArguments.scala:14)
    at scala.collection.LinearSeqOptimized.foldLeft (LinearSeqOptimized.scala:126)
    at scala.collection.LinearSeqOptimized.foldLeft$ (LinearSeqOptimized.scala:122)
    at scala.collection.immutable.List.foldLeft (List.scala:89)
    at io.github.evis.scalafix.maven.plugin.phases.BuildScalafixArguments.buildScalafixArguments (BuildScalafixArguments.scala:14)
    at io.github.evis.scalafix.maven.plugin.phases.BuildScalafixArguments.buildScalafixArguments$ (BuildScalafixArguments.scala:10)
    at io.github.evis.scalafix.maven.plugin.mojo.package$.buildScalafixArguments (package.scala:5)
    at io.github.evis.scalafix.maven.plugin.phases.Run.run (Run.scala:14)
    at io.github.evis.scalafix.maven.plugin.phases.Run.run$ (Run.scala:12)
    at io.github.evis.scalafix.maven.plugin.mojo.package$.run (package.scala:5)
    at io.github.evis.scalafix.maven.plugin.mojo.ScalafixMojo.execute (ScalafixMojo.scala:101)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

Maven

```
mvn --version
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T14:41:47-04:00)
Maven home: /opt/maven
Java version: 1.8.0_201, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-65-generic", arch: "amd64", family: "unix"
```

--- a/src/main/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParam.scala
+++ b/src/main/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParam.scala
@@ -25,7 +25,7 @@ object PluginsParam {
 
   @SuppressWarnings(Array("scalafix:DisableSyntax.asInstanceOf"))
   private def config(plugin: Plugin): Option[Xpp3Dom] = {
-    Try(Some(plugin.getConfiguration.asInstanceOf[Xpp3Dom])).getOrElse(None)
+    Try(Option(plugin.getConfiguration.asInstanceOf[Xpp3Dom])).getOrElse(None)
   }
 
   private def compilerArgs(config: Xpp3Dom): List[String] = {

--- a/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
+++ b/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
@@ -1,0 +1,37 @@
+package io.github.evis.scalafix.maven.plugin.params
+
+import java.io.StringReader
+
+import io.github.evis.scalafix.maven.plugin.{BaseSpec, ScalafixArgumentsBuilder}
+import org.apache.maven.model.Plugin
+import org.codehaus.plexus.util.xml.Xpp3DomBuilder
+
+import scala.collection.JavaConverters._
+
+class PluginsParamSpec extends BaseSpec {
+
+  def scalaPlugin(args: List[String]): Plugin = {
+    val sp = new Plugin()
+    sp.setGroupId("net.alchim31.maven")
+    sp.setArtifactId("scala-maven-plugin")
+    if (args.nonEmpty) {
+      val argsLine = args.mkString("<arg>", "</arg><arg>", "</arg>")
+      val configurationString = s"""<configuration><args>$argsLine</args></configuration>"""
+      sp.setConfiguration(
+        Xpp3DomBuilder.build(new StringReader(configurationString))
+      )
+    }
+    sp
+  }
+
+  "PluginsParam" should "return scalacOptions if they exist" in {
+    PluginsParam(List(scalaPlugin("-Xlint" :: "-Ypartial-unification" :: Nil)).asJava)(ScalafixArgumentsBuilder())
+      .scalacOptions shouldBe List("-Xlint", "-Ypartial-unification")
+  }
+
+  it should "return no scalacOptions if they are not set" in {
+    PluginsParam(List(scalaPlugin(Nil)).asJava)(ScalafixArgumentsBuilder())
+      .scalacOptions shouldBe Nil
+  }
+
+}

--- a/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
+++ b/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
@@ -16,7 +16,8 @@ class PluginsParamSpec extends BaseSpec {
     sp.setArtifactId("scala-maven-plugin")
     if (args.nonEmpty) {
       val argsLine = args.mkString("<arg>", "</arg><arg>", "</arg>")
-      val configurationString = s"""<configuration><args>$argsLine</args></configuration>"""
+      val configurationString =
+        s"""<configuration><args>$argsLine</args></configuration>"""
       sp.setConfiguration(
         Xpp3DomBuilder.build(new StringReader(configurationString))
       )
@@ -25,13 +26,15 @@ class PluginsParamSpec extends BaseSpec {
   }
 
   "PluginsParam" should "return scalacOptions if they exist" in {
-    PluginsParam(List(scalaPlugin("-Xlint" :: "-Ypartial-unification" :: Nil)).asJava)(ScalafixArgumentsBuilder())
-      .scalacOptions shouldBe List("-Xlint", "-Ypartial-unification")
+    PluginsParam(
+      List(scalaPlugin("-Xlint" :: "-Ypartial-unification" :: Nil)).asJava)(
+      ScalafixArgumentsBuilder()).scalacOptions shouldBe List(
+      "-Xlint",
+      "-Ypartial-unification")
   }
 
   it should "return no scalacOptions if they are not set" in {
-    PluginsParam(List(scalaPlugin(Nil)).asJava)(ScalafixArgumentsBuilder())
-      .scalacOptions shouldBe Nil
+    PluginsParam(List(scalaPlugin(Nil)).asJava)(ScalafixArgumentsBuilder()).scalacOptions shouldBe Nil
   }
 
 }

--- a/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
+++ b/src/test/scala/io/github/evis/scalafix/maven/plugin/params/PluginsParamSpec.scala
@@ -2,13 +2,12 @@ package io.github.evis.scalafix.maven.plugin.params
 
 import java.io.StringReader
 
-import io.github.evis.scalafix.maven.plugin.{BaseSpec, ScalafixArgumentsBuilder}
 import org.apache.maven.model.Plugin
 import org.codehaus.plexus.util.xml.Xpp3DomBuilder
 
 import scala.collection.JavaConverters._
 
-class PluginsParamSpec extends BaseSpec {
+class PluginsParamSpec extends ParamSpec {
 
   def scalaPlugin(args: List[String]): Plugin = {
     val sp = new Plugin()
@@ -27,14 +26,13 @@ class PluginsParamSpec extends BaseSpec {
 
   "PluginsParam" should "return scalacOptions if they exist" in {
     PluginsParam(
-      List(scalaPlugin("-Xlint" :: "-Ypartial-unification" :: Nil)).asJava)(
-      ScalafixArgumentsBuilder()).scalacOptions shouldBe List(
+      List(scalaPlugin("-Xlint" :: "-Ypartial-unification" :: Nil)).asJava).applied.scalacOptions shouldBe List(
       "-Xlint",
       "-Ypartial-unification")
   }
 
   it should "return no scalacOptions if they are not set" in {
-    PluginsParam(List(scalaPlugin(Nil)).asJava)(ScalafixArgumentsBuilder()).scalacOptions shouldBe Nil
+    PluginsParam(List(scalaPlugin(Nil)).asJava).applied.scalacOptions shouldBe Nil
   }
 
 }

